hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirRenameOp.java

```diff
@@ -417,12 +434,32 @@ static RenameResult unprotectedRenameTo(FSDirectory fsd,
 
     // Ensure dst has quota to accommodate rename
     verifyFsLimitsForRename(fsd, srcIIP, dstIIP);
-    verifyQuotaForRename(fsd, srcIIP, dstIIP);
+    QuotaCounts srcPolicyCounts = new QuotaCounts.Builder(true).build();
+    QuotaCounts dstPolicyCounts = new QuotaCounts.Builder(true).build();
+    boolean srcIIPIsQuota = FSDirectory.verifyIsQuota(
+        srcIIP, srcIIP.length() - 1);
+    boolean dstIIPIsQuota = FSDirectory.verifyIsQuota(
+        dstIIP, dstIIP.length() - 1);
+    if (srcIIPIsQuota || dstIIPIsQuota) {
+      if (dstParent.getStoragePolicyID() ==
+          srcIIP.getLastINode().getStoragePolicyID()) {
+        srcPolicyCounts = srcIIP.getLastINode().computeQuotaUsage(bsps);
```

Comment:
line 446 and 449 are the same. You may move them to line 444.

Reply:
Thanks a lot for your review . I will update it .